### PR TITLE
[ISSUE #166] Provide updated Runtime/Docker deployment and compilation docs

### LINKED ISSUE #166: [Doc] Provide updated Runtime/Docker deployment and compilation docs
Related PR: https://github.com/apache/eventmesh/pull/4675

### PR DESCRIPTION
**Pull request by Pil0tXia**
Fixes #166.

The documentation for Runtime and Runtime Docker has not been updated for a long time, containing numerous outdated commands and incorrect descriptions. This PR addresses all of these issues and optimizes the structure of the Runtime documentation, making it more user-friendly, web-friendly, and comprehensible.

Preview markdown web pages:

- [EventMesh Runtime with Docker](https://github.com/Pil0tXia/eventmesh-site/blob/pil0txia_doc_166/i18n/zh/docusaurus-plugin-content-docs/current/instruction/04-runtime-with-docker.md)
- [EventMesh Runtime Quick Start](https://github.com/Pil0tXia/eventmesh-site/blob/pil0txia_doc_166/i18n/zh/docusaurus-plugin-content-docs/current/instruction/03-runtime.md)

Additionally, this PR optimizes the following:
- Supplementary fixes for #162, correcting the copyright desc and the "incubator" word on the Chinese website.
- The v1.10.0 documentation still used the v1.9.0 documentation; now the v1.10.0 documentation is consistent with Next.
- Updated multiple version numbers, such as in the Release Notes on the Roadmap page.
- Synchronized all connector documentation.
- Improved the Connector Implementation Status table, moved it to the Roadmap for better visibility. Added the Event Store Implementation Status.
- Made less important Eclipse documentation less prominent.
- Added code highlighting for the following languages to the code block: bash, gradle, groovy, properties, json, protobuf, sql, antlr4, log, rust.
- Upgraded docusaurus from 2.4.1 to 2.4.3 (no breaking change) and node from 16 to 18 (16 has End of Life).
- Many other minor changes.

In the next PR, I am working on updating the Roadmap and addressing image relative path issues.

### CODE REVIEW

File: docs/roadmap.md [hunk 1]
```diff
@@ -48,3 +48,49 @@ The development roadmap of Apache EventMesh is an overview of the planned featur
...
+|                     ChatGPT                      |    ⬜    |    ⬜    |
+|        [Slack](./design-document/connect/slack-connector)        |    ⬜    |    ✅    |
+|         More connectors will be added...         |   N/A   |   N/A   |
+
+## Event Store Implementation Status
+
+| Service / Middleware | Ingress | Topic Management |
+|:------------------------:|:------:|:------:|
+| RocketMQ | ✅ | ✅ |
+| Kafka | ✅ | ✅ |
```
Comment (pandaapo) [inline docs/roadmap.md:91]: Does 'Topic Management' here refer to management in eventmesh-admin or eventmesh-dashboard (dev)? If so, in previous eventmesh-admin, topic management of RocketMQ seems to be an empty implementation and Kafka doesn't seem to have it implemented, and in current eventmesh-dashboard, they have been deleted.

File: i18n/zh/docusaurus-plugin-content-docs/version-v1.10.0/design-document/01-spi.md [hunk 2]
```diff
@@ -91,15 +91,15 @@ public interface MeshMQProducer extends Producer {
 }
 ```
 
-eventmesh-connector-rocketmq模块中包含采用rocketmq的具体实现方式RocketMQProducerImpl。
+eventmesh-connector-rocketmq 模块中包含采用 rocketmq 的具体实现方式 RocketMQProducerImpl。
```
Comment (pandaapo) [inline i18n/zh/docusaurus-plugin-content-docs/version-v1.10.0/design-document/01-spi.md:94]: These should be `eventmesh-storage-rocketmq`.

File: docs/instruction/03-runtime.md [hunk 3]
```diff
@@ -1,215 +1,270 @@
...
 
-### 2.3 Deploy
+### 3.3 Project Structure Explanation
 
-Edit the `eventmesh.properties` to change the configuration (e.g. TCP port, client blacklist) of EventMesh Runtime. The executable binaries contain all plugins in the bundle, thus there's no need to build them from source code.
+| Module Name                | Description                          |
+| -------------------------- | ------------------------------------ |
+| eventmesh-common           | EventMesh common classes and methods module |
+| eventmesh-connector-api    | EventMesh Connector plugin interface definition module |
+| eventmesh-connector-plugin | EventMesh Connector plugin module    |
```
Comment (pandaapo) [inline docs/instruction/03-runtime.md:203]: The `eventmesh-connector-api` and `eventmesh-connector-plugin` modules are no longer present in the project.

File: docs/instruction/03-runtime.md [hunk 4]
```diff
@@ -1,215 +1,270 @@
...
+| eventmesh-spi              | EventMesh SPI loading module          |
 
-```console
-vim conf/eventmesh.properties
-```
+> Plugin modules follow the SPI specification defined by EventMesh, and custom SPI interfaces need to be annotated with `@EventMeshSPI`.
+>
+> Plugin instances need to be configured in the corresponding module under the `/main/resources/META-INF/eventmesh` directory with a mapping file for interface and implementation classes. The file name is the fully qualified class name of the SPI interface.
+>
+> The content of the file is the mapping from the plugin instance name to the plugin instance. For details, refer to the `eventmesh-connector-rocketmq` plugin module.
```
Comment (pandaapo) [inline docs/instruction/03-runtime.md:213]: eventmesh-connector-rocketmq does not run in the plugin way described above.

File: docs/instruction/03-runtime.md [hunk 5]
```diff
@@ -1,215 +1,270 @@
...
+
+EventMesh has an SPI mechanism that allows EventMesh to discover and load plugins. There are two ways to install plugins:
+
+- Classpath loading: During local development, you can add dependencies in the `build.gradle` of the `eventmesh-starter` module. For example, to add the Kafka Connector plugin:
+
+```gradle
+dependencies {
+   implementation project(":eventmesh-runtime")
+    // Example: Add the Kafka Connector plugin
+   implementation project(":eventmesh-connectors:eventmesh-connector-kafka")
```
Comment (pandaapo) [inline docs/instruction/03-runtime.md:227]: eventmesh-connectors do not run in the plugin way described above.

File: docs/instruction/03-runtime.md [hunk 6]
```diff
@@ -1,215 +1,270 @@
...
 
-View the output log:
+EventMesh will load the plugins by default from the `dist/plugin` directory. You can change the plugin directory using `-DeventMeshPluginDir=your_plugin_directory`. The plugin instances needed at runtime can be configured in the `confPath` directory in the `eventmesh.properties` file. For example, by setting the following, you declare the use of the RocketMQ Connector plugin:
 
-```console
-cd /root/apache-eventmesh-1.9.0/logs
-tail -f eventmesh.out
+```properties
+# connector plugin
+eventMesh.connector.plugin.type=rocketmq
```
Comment (pandaapo) [inline docs/instruction/03-runtime.md:245]: eventmesh-connector-rocketmq does not run in the plugin way described above.

File: i18n/zh/docusaurus-plugin-content-docs/current/instruction/03-runtime.md [hunk 7]
```diff
@@ -1,216 +1,271 @@
-# EventMesh Runtime 快速入门说明
+# EventMesh Runtime 快速开始
 
 EventMesh Runtime 是 EventMesh 集群中有状态的 Mesh 节点，负责 Source Connector 与 Sink Connector 之间的事件传输，并可以使用 Event Store 作为事件的存储队列。
 
-![EventMesh Runtime](/images/design-document/runtime.png)
+![EventMesh Runtime](../../../../../static/images/design-document/runtime.png)
```
Comment (pandaapo) [inline i18n/zh/docusaurus-plugin-content-docs/current/instruction/03-runtime.md:5]: Isn't using the writing method of `/static/images/...` simpler?

File: versioned_docs/version-v1.10.0/sdk-java/01-intro.md [hunk 8]
```diff
@@ -30,12 +30,10 @@ To install EventMesh SDK for Java with Maven, declare `org.apache.eventmesh:even
   <dependency>
     <groupId>org.apache.eventmesh</groupId>
     <artifactId>eventmesh-sdk-java</artifactId>
-    <version>1.4.0</version>
+    <version>1.9.0</version>
```
Comment (pandaapo) [inline versioned_docs/version-v1.10.0/sdk-java/01-intro.md:33]: This should be `1.10.0`.

**Review by pandaapo — COMMENTED** [review]
Due to my limited personal abilities, in too many changes to files, I did not review the modifications to historical versions, workflow and schema that I do not understand, etc. And I did not comment on all the similar issues one by one.

由于个人能力有限，改动文件太多，对于历史版本的修改、workflow和schema（我不懂）的修改等等内容，我没有review；对于类似的问题我也没有逐个全部标记。

File: docs/roadmap.md (hunk 1 above)
Comment (Pil0tXia) [reply docs/roadmap.md:91]: The "Topic Management" refers to the `admin` package of each storage-plugin, which is used in the "Topic" page of the original eventmesh-dashboard:main.

File: i18n/zh/docusaurus-plugin-content-docs/current/instruction/03-runtime.md (hunk 7 above)
Comment (Pil0tXia) [reply i18n/zh/docusaurus-plugin-content-docs/current/instruction/03-runtime.md:5]: Using `/static/images/...` will cause docusaurus build error:

```
[INFO] [zh] Creating an optimized production build...


Error: Image static/static/images/design-document/runtime.png used in i18n/zh/docusaurus-plugin-content-docs/current/instruction/03-runtime.md not found.
    at async Promise.all (index 0)
[ERROR] Client bundle compiled with errors therefore further build is impossible.
```

File: versioned_docs/version-v1.10.0/sdk-java/01-intro.md (hunk 8 above)
Comment (Pil0tXia) [reply versioned_docs/version-v1.10.0/sdk-java/01-intro.md]: Fixed in https://github.com/apache/eventmesh-site/pull/170/commits/dad1f5807f4146a7b06f8eb45e8cb350f5e7f6ae

File: docs/instruction/03-runtime.md (hunk 4 above)
Comment (Pil0tXia) [reply docs/instruction/03-runtime.md]: The connector mentioned in this document is now referred to as a storage plugin. Fixed in https://github.com/apache/eventmesh-site/pull/170/commits/dad1f5807f4146a7b06f8eb45e8cb350f5e7f6ae

File: docs/instruction/03-runtime.md (hunk 5 above)
Comment (Pil0tXia) [reply docs/instruction/03-runtime.md]: Fixed in https://github.com/apache/eventmesh-site/pull/170/commits/dad1f5807f4146a7b06f8eb45e8cb350f5e7f6ae.

File: docs/instruction/03-runtime.md (hunk 6 above)
Comment (Pil0tXia) [reply docs/instruction/03-runtime.md]: Fixed in https://github.com/apache/eventmesh-site/pull/170/commits/dad1f5807f4146a7b06f8eb45e8cb350f5e7f6ae.

File: docs/instruction/03-runtime.md (hunk 3 above)
Comment (Pil0tXia) [reply docs/instruction/03-runtime.md]: Updated in https://github.com/apache/eventmesh-site/pull/170/commits/dad1f5807f4146a7b06f8eb45e8cb350f5e7f6ae.

**Comment by Pil0tXia** [comment]
This PR does not have any content modifications to the documentation of historical versions, workflows, and schemas.

Speaking of historical versions, there are significant issues with them currently. Multiple versions are almost identical, and they do not align with the document structure and content in the Git repository. I first mentioned this issue in https://github.com/apache/eventmesh-site/pull/132#discussion_r1322364177.

**Comment by pandaapo** [comment]
For the contents that I can review, they are already OK for me.

**Review by xwm1992 — APPROVED** [review]

**Pull request merged**